geth/jail/internal/loop/loop.go

```diff
@@ -135,19 +131,24 @@ func (l *Loop) processTask(t Task) error {
 
 // Run handles the task scheduling and finalisation.
 // It runs infinitely waiting for new tasks.
-func (l *Loop) Run() error {
-	for t := range l.ready {
-		if t == nil {
-			continue
-		}
-
-		err := l.processTask(t)
-		if err != nil {
-			// TODO(divan): do we need to report
-			// errors up to the caller?
-			// Ignoring for now, as loop
-			// should keep running.
-			continue
+func (l *Loop) Run(ctx context.Context) error {
+	for {
+		select {
+		case t := <-l.ready:
+			if t == nil {
+				continue
+			}
+
+			err := l.processTask(t)
+			if err != nil {
+				// TODO(divan): do we need to report
+				// errors up to the caller?
+				// Ignoring for now, as loop
+				// should keep running.
```

Comment:
Very good question. How are failing tasks handled?

Reply:
They are not. It's responsibility of the caller to handle error in the callback functions they launch via `setTimeout`.

Reply:
Ah, IC, thanks. So the caller gets notified about a timeout but not about why this happens? and the caller is not interested in the cause?

Reply:
@themue that's a good questions, but they are unrelated to this PR. This TODO was written long before, but to answer it we should brainstorm it with status-react devs, to understand properly expectations of dApps users or whoever uses Jail.